File: backend/app/v1/schemas/job_position.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class JobPositionBase(BaseModel):
    name: str

class JobPositionCreate(JobPositionBase):
    pass

class JobPositionUpdate(BaseModel):
    name: str | None = None

class JobPositionRead(JobPositionBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/jobPosition.ts
import type { JobPositionCreateFormValues,JobPositionUpdateFormValues} from "@/schemas/jobPosition";
/**
 * Job position returned from read operations.
 */
export interface JobPositionRead {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
}

/**
 * Payload for creating a new job position.
 */
export type JobPositionCreate= JobPositionCreateFormValues

/**
 * Payload for updating an existing job position.
 */
export type JobPositionUpdate =JobPositionUpdateFormValues
